File: packages/js-sdk/src/regex/detectors/be/enterprise-number.ts
// Belgian Enterprise Number (BCE/KBO) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { beEnterpriseChecksum } from '../../validators'

class BeEnterpriseNumberDetector extends RegexDetector {
  entityType = EntityType.BE_ENTERPRISE_NUMBER
  score = 0.85
  contextKeywords = [
    'bce',
    'kbo',
    'enterprise number',
    'ondernemingsnummer',
    'numero entreprise',
  ]
  contextRequired = true
  contextWindow = 50

  pattern = /\b[01]\d{3}[. ]?\d{3}[. ]?\d{3}\b/g
  validator = beEnterpriseChecksum
}

registerRegion('be', BeEnterpriseNumberDetector)

export { BeEnterpriseNumberDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/be/enterprise_number.py
"""Belgian Enterprise Number (BCE/KBO) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import be_enterprise_checksum


@register_region("be")
class EnterpriseNumberDetector(RegexDetector):
    entity_type = EntityType.BE_ENTERPRISE_NUMBER
    score = 0.85
    context_keywords = [
        "bce", "kbo", "enterprise number", "ondernemingsnummer",
        "numero entreprise",
    ]
    context_required = True
    context_window = 50

    pattern = re.compile(
        r"\b[01]\d{3}[. ]?\d{3}[. ]?\d{3}\b"
    )
    validator = staticmethod(be_enterprise_checksum)
